### DISCUSSION TITLE
Save session each time a tab is open, closed or updated, to better organize and save resources avoiding session duplicates

### DISCUSSION THREAD

**Discussion by leoncrf** [sienori/Tab-Session-Manager#1157] [Ideas]
I believe some people use self save session every 1 minute to lose anything, but this accumulates a large number of saved sessions, and limiting the number of saved sessions would not be so useful either because we still have exactly equal saved sessions, while we would also lose older sessions that were discarded by limit of saved sessions.

Save the session each time a tab is open, closed or updated would make the amount of saved sessions more organized, avoid duplicates and probably populate some amount of memory.

**Comment by Bryan-Longworth** [comment]
I'm having the same problem. I have to constantly deleted new sessions that are created while the existing session isn't updated with new tabs added or closed tabs removed. I have used Simple Tab Groups which does successfully update existing sessions or groups with new or removed tabs, but I'm trying Tab Session Manager because STG opens every tab when I open Firefox even though I only intend to open one tab group. It also temporarily adds a closed group to another open group, and if you don't catch that and close that group, it will have the tabs in that group plus the previously closed group.
